posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,311 @@
+
+The transformer can be run using custom rules by specifying additional options on the command line.  The additional options specify local rules files that the transformer is to be used instead of the default rules files.
+
+For example, Open Liberty has more up-to-date rules for performing transformations.  To use the Open Liberty rules, download these from the Open Liberty GIT repository: link:https://github.com/OpenLiberty/open-liberty/tree/integration/dev/wlp-jakartaee-transform/rules[Open Liberty Rules].
+
+Add three command line options to the transformer command line invocation to pull in the more up-to-date rules data:
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location> \
+    -tr ${rules}/jakarta-renames.properties \
+    -td ${rules}/jakarta-direct.properties \
+    -tf ${rules}/jakarta-xml-master.properties
+--
+
+(These three command line options are most important.  While other command line options are available, for example, for transforming bundle archives, and for more finely selecting which target files are to be transformed, those other options are not usually needed.)
+
+The target location can be a packaged application archive (EAR, WAR, RAR, or JAR file) or can be a directory containing the fully or partially expanded contents of one of these archives.  The target can also be a directory containing Java source, or a directory containing several application archives.  All of the artifacts in the target location are processed by the transformer.
+
+=== More about transforming server configurations
+
+Transformation of a server configuration is only necessary when "javax" prefixed package names are exposed in the server configuration.  One case is JMS activation specification elements.  For example:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="javax.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+This must be transformed to:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="jakarta.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+== Updating features in server configurations:
+
+When running a Jakarta EE 9 enabled application using Open Liberty, Jakarta EE 9 features must be specified in the server configuration.  In many cases, Jakarta EE 9 features use the same short names as the corresponding Java / Jakarta EE features.  The Jakarta EE 9 features simply update the feature version.  In other cases, both the feature short name and the feature version must be updated.
```

Comment:
For the last two sentences I suggest:

In those cases, the Jakarta EE 9 features simple have an updated feature version.  In other cases both the feature short name and the feature version are updated.

Reply:
I had a typo.  It should have been `simply` instead of `simple`

Reply:
Fixed.